src/Core/Notifications/IDisabledNotificationsStorage.cs

```diff
@@ -32,15 +40,95 @@ public interface IDisabledNotificationsStorage
+        private readonly IVsVersionProvider vsVersionProvider;
+        private readonly IFileSystem fileSystem;
+        private readonly ILogger logger;
+
+        private string filePath => GetFilePath();
+
+        private DisabledNotifications disabledNotifications = null;
+        private readonly object lockObject = new object();
+
+        [ImportingConstructor]
+        public DisabledNotificationsStorage(IVsVersionProvider vsVersionProvider, ILogger logger) : this(vsVersionProvider, logger, new FileSystem())
         {
-            // todo
+
         }
 
+        internal /*for testing*/ DisabledNotificationsStorage(IVsVersionProvider vsVersionProvider, ILogger logger, IFileSystem fileSystem)
+        {
+            this.vsVersionProvider = vsVersionProvider;
+            this.fileSystem = fileSystem;
+            this.logger = logger;
+        }
+
+        public void DisableNotification(string id)
+        {
+            lock (lockObject)
+            {
+                if (IsNotificationDisabled(id))
+                {
+                    return;
+                }
+                if (disabledNotifications == null)
+                {
+                    return;
+                }
+
+                disabledNotifications.AddNotification(id);
+                SaveNotifications();
+            }
+        }
         public bool IsNotificationDisabled(string id)
         {
-            // todo
-            return false;
+            lock (lockObject)
+            {
+                if (disabledNotifications == null)
+                {
+                    disabledNotifications = ReadDisabledNotifications();
+                }
+                if (disabledNotifications == null)
+                {
+                    logger.WriteLine("Couldn't find disabled notification config. Notification will be showed");
+                    return false;
+                }
+                return disabledNotifications.Notifications.Any(n => n.Id == id);
+            }
+        }
+
+        private DisabledNotifications ReadDisabledNotifications()
+        {
+            try
+            {
+                var fileContent = fileSystem.File.ReadAllText(filePath);
+                return JsonConvert.DeserializeObject<DisabledNotifications>(fileContent);
+            }
+            catch (Exception ex)
+            {
+                logger.WriteLine($"Failed to read disabled notification config from disk. Error:{ex.Message}");                
+            }
+            return null;
+        }
+
+        private void SaveNotifications()
+        {
+            try
+            {
+                var fileContent = JsonConvert.SerializeObject(disabledNotifications);
+                fileSystem.File.WriteAllText(filePath, fileContent);
+            }
+            catch (Exception ex)
+            {
+                logger.WriteLine($"Failed to save disabled notification config to disk. Error:{ex.Message}");
```

Comment:
All non-debug messages should be in a resource file.

Also, the message should be prefixed with `[Notifications] `.

Same for the other error message.